Toolkit/ArcGISToolkit/TimeSlider.swift

```diff
@@ -0,0 +1,2539 @@
+            }
+            
+            let thumbStartTimeLabelY = lowerThumbLayer.frame.minY - currentExtentStartTimeLabel.frame.height - labelPadding
+            let tickLayerStartTimeLabelY = tickMarkLayer.frame.minY - currentExtentStartTimeLabel.frame.height - labelPadding
+            let startTimeLabelY = min(thumbStartTimeLabelY, tickLayerStartTimeLabelY)
+            currentExtentStartTimeLabel.frame = CGRect(x: startTimeLabelX, y: startTimeLabelY, width: startTimeLabelSize.width, height: startTimeLabelSize.height)
+        }
+        else {
+            currentExtentStartTimeLabel.string = ""
+            currentExtentStartTimeLabel.isHidden = true
+        }
+        
+        // Update upper label
+        if let endTime = currentExtentEndTime, isRangeEnabled, fullExtent != nil  {
+            let endTimeString = currentExtentDateFormatter.string(from: endTime)
+            currentExtentEndTimeLabel.string = endTimeString
+            let endTimeLabelSize: CGSize = endTimeString.size(withAttributes: [kCTFontAttributeName as NSAttributedStringKey: currentExtentLabelFont])
+            var endTimeLabelX = upperThumbLayer.frame.midX - endTimeLabelSize.width / 2
+            currentExtentEndTimeLabel.isHidden = !isSliderVisible
+            if endTimeLabelX < bounds.origin.x + labelSidePadding && currentExtentStartTime == currentExtentEndTime {
+                endTimeLabelX = bounds.origin.x + labelSidePadding
+                currentExtentEndTimeLabel.isHidden = true
+            }
+            else if endTimeLabelX + endTimeLabelSize.width > bounds.maxX - labelSidePadding {
+                endTimeLabelX = bounds.maxX - endTimeLabelSize.width - labelSidePadding
+                currentExtentEndTimeLabel.isHidden = true
+            }
+            else if endTimeLabelX < currentExtentStartTimeLabel.frame.origin.x + currentExtentStartTimeLabel.frame.width {
+                endTimeLabelX = currentExtentStartTimeLabel.frame.origin.x + currentExtentStartTimeLabel.frame.width + paddingBetweenLabels
+            }
+            
+            let thumbEndTimeLabelY = upperThumbLayer.frame.minY - currentExtentEndTimeLabel.frame.height - labelPadding
+            let tickLayerEndTimeLabelY = tickMarkLayer.frame.minY - currentExtentEndTimeLabel.frame.height - labelPadding
+            let endTimeLabelY = min(thumbEndTimeLabelY, tickLayerEndTimeLabelY)
+            currentExtentEndTimeLabel.frame = CGRect(x: endTimeLabelX, y: endTimeLabelY, width: endTimeLabelSize.width, height: endTimeLabelSize.height)
+        }
+        else {
+            currentExtentEndTimeLabel.string = ""
+            currentExtentEndTimeLabel.isHidden = true
+        }
+        
+        // Commit the transaction
+        CATransaction.commit()
+    }
+    
+    private func positionTickMarks() {
+        //
+        // Bail out if time steps are not available
+        guard let timeSteps = timeSteps, timeSteps.count > 0, isSliderVisible else {
```

Comment:
`!timeSteps.isEmpty` is preferred to `timeSteps.count > 0` as it is guaranteed to be O(1).

> When you need to check whether your collection is empty, use the isEmpty property instead of checking that the count property is equal to zero. For collections that don’t conform to RandomAccessCollection, accessing the count property iterates through the elements of the collection.

Reply:
fixed